### ISSUE TITLE
windows version's suricataupdate.exe can't use local rules

### ISSUE THREAD

**Issue by mikal70** [brimdata/brimcap#259]
Hi Sir:
as title, when i use windows version's brimcap, i want to update suricatarules by suricateupdate.exe,
but it can't accept any command or parameter,can it fix?
suricataupdate work well in kali.
below pic is in windows.
![image](https://user-images.githubusercontent.com/34530371/177483382-58be299d-9a88-42dc-9444-ea4b641f1b61.png)

and this is in kali
![image](https://user-images.githubusercontent.com/34530371/177483538-5839a985-5869-44cc-afdc-83854fb2e0c4.png)

thx!

**Comment by philrz** [comment]
@mikal70: Yes, I see the problem. On Linux (and macOS) the `suricataupdater` is a shell script that includes `"$@"` at the end of its `suricata-update` command line, which allows for passing along additional options such as the `--local` you're doing here. On Windows the shell isn't rich enough to do everything that was needed to invoke `suricata-update` correctly, so a [wrapper](https://github.com/brimdata/build-suricata/blob/master/go/updater/main.go) was written in Go that tries to be similar in spirit, but is shipped as an executable. Unfortunately that wrapper was only invoking a static command line and not doing the equivalent of the `"$@"` that the shell wrapper was doing, hence the limitation you're seeing.

I've taken a shot at enhancing the Go wrapper and have attached it here as [suricataupdater.zip](https://github.com/brimdata/brimcap/files/9074112/suricataupdater.zip). I know firewalls often freak out if they see Windows executables being downloaded, so once you unpack the ZIP, rename its extension from `.txt` to `.exe`.

Below I'll paste some test output indicating it seems to be working for me. Before starting the test, I've got the following sitting in my `C:\temp`:

* The unpacked Brimcap release https://github.com/brimdata/brimcap/releases/download/v1.2.0/brimcap-v1.2.0.windows-amd64.zip
* The enhanced `suricataupdater.exe` linked above
* A `local.rules` file with the following contents based on what I found in the [Suricata docs](https://suricata.readthedocs.io/en/latest/rule-management/adding-your-own-rules.html):

      alert http any any -> any any (msg:"Do not read gossip during work"; content:"Scarlett"; nocase; classtype:policy-violation; sid:9999999; rev:1;)

Here's the test output:

```
C:\temp>cd \temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata

C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata>copy \temp\suricataupdater.exe .
Overwrite .\suricataupdater.exe? (Yes/No/All): yes
        1 file(s) copied.

C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata>mkdir share\suricata\CVE_RULES

C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata>copy \temp\local.rules share\suricata\CVE_RULES\local.rules
        1 file(s) copied.

C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata>.\suricataupdater.exe --local \temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\share\suricata\CVE_RULES
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Loading C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\var\lib\suricata\update.yaml←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Found Suricata version 5.0.3 at C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\bin\suricata.exe.←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Loading C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\brim-conf.yaml←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Disabling rules for protocol modbus←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Disabling rules for protocol dnp3←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Disabling rules for protocol enip←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- No sources configured, will use Emerging Threats Open←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Last download less than 15 minutes ago. Not downloading https://rules.emergingthreats.net/open/suricata-5.0.3/emerging.rules.tar.gz.←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Loading local file \temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\share\suricata\CVE_RULES\local.rules←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Loading distribution rule file C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\share\suricata\rules\app-layer-events.rules←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Loading distribution rule file C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\share\suricata\rules\decoder-events.rules←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Loading distribution rule file C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\share\suricata\rules\dhcp-events.rules←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Loading distribution rule file C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\share\suricata\rules\dnp3-events.rules←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Loading distribution rule file C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\share\suricata\rules\dns-events.rules←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Loading distribution rule file C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\share\suricata\rules\files.rules←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Loading distribution rule file C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\share\suricata\rules\http-events.rules←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Loading distribution rule file C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\share\suricata\rules\ipsec-events.rules←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Loading distribution rule file C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\share\suricata\rules\kerberos-events.rules←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Loading distribution rule file C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\share\suricata\rules\modbus-events.rules←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Loading distribution rule file C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\share\suricata\rules\nfs-events.rules←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Loading distribution rule file C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\share\suricata\rules\ntp-events.rules←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Loading distribution rule file C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\share\suricata\rules\smb-events.rules←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Loading distribution rule file C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\share\suricata\rules\smtp-events.rules←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Loading distribution rule file C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\share\suricata\rules\stream-events.rules←[0m
←[32m8/7/2022 -- 10:52:22←[0m - <←[33mInfo←[0m> -- Loading distribution rule file C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\share\suricata\rules\tls-events.rules←[0m
←[32m8/7/2022 -- 10:52:23←[0m - <←[33mInfo←[0m> -- Ignoring file rules/emerging-deleted.rules←[0m
←[32m8/7/2022 -- 10:52:25←[0m - <←[33mInfo←[0m> -- Loaded 30169 rules.←[0m
←[32m8/7/2022 -- 10:52:25←[0m - <←[33mInfo←[0m> -- Disabled 14 rules.←[0m
←[32m8/7/2022 -- 10:52:25←[0m - <←[33mInfo←[0m> -- Enabled 0 rules.←[0m
←[32m8/7/2022 -- 10:52:25←[0m - <←[33mInfo←[0m> -- Modified 0 rules.←[0m
←[32m8/7/2022 -- 10:52:25←[0m - <←[33mInfo←[0m> -- Dropped 0 rules.←[0m
←[32m8/7/2022 -- 10:52:25←[0m - <←[33mInfo←[0m> -- Enabled 147 rules for flowbit dependencies.←[0m
←[32m8/7/2022 -- 10:52:25←[0m - <←[33mInfo←[0m> -- Backing up current rules.←[0m
←[32m8/7/2022 -- 10:52:28←[0m - <←[33mInfo←[0m> -- Writing rules to C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\var\lib\suricata\rules\suricata.rules: total: 30169; enabled: 22723; added: 1; removed 0; modified: 14←[0m
←[32m8/7/2022 -- 10:52:29←[0m - <←[33mInfo←[0m> -- Writing C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata\var\lib\suricata\rules\classification.config←[0m
←[32m8/7/2022 -- 10:52:29←[0m - <←[33mInfo←[0m> -- Skipping test, disabled by configuration.←[0m
←[32m8/7/2022 -- 10:52:29←[0m - <←[33mInfo←[0m> -- Done.←[0m

C:\temp\brimcap-v1.2.0.windows-amd64\brimcap\suricata>grep gossip var\lib\suricata\rules\suricata.rules
alert http any any -> any any (msg:"Do not read gossip during work"; content:"Scarlett"; nocase; classtype:policy-violation; sid:9999999; rev:1;)
```

If you can confirm that this `suricataupdate.exe` also works for your use case, I can go ahead with getting the code changes reviewed and baked into a future Brimcap release.

**Comment by philrz** [comment]
@mikal70: Glad to hear it's working for you. I've put up a change at brimdata/build-suricata#63 that, once reviewed/merged, I can bundle into a future Brimcap release. Alas, it looks like our test automation has bit-rotted a bit so I have other things to fix first. Since you're up & running with that test executable hopefully you're ok in the meantime. I'll update here when a Brimcap release has the final fix.

**Comment by philrz** [comment]
This issue is fixed in the Suricata that's bundled with the recent GA Brimcap release [v1.3.0](https://github.com/brimdata/brimcap/releases/tag/v1.3.0), which will be included in the upcoming Zui app v1.0.0 GA release.

**Comment by philrz** [comment]
The [Zui v1.0.0](https://github.com/brimdata/zui/releases/tag/v1.0.0) release went GA on Monday and includes this fix.
